### PR TITLE
specify allow(dead_code) to be only on windows in the Args struct

### PR DESCRIPTION
**Pull request by w1kee**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -159,7 +159,7 @@ static ARGS: Lazy<Args> = Lazy::new(|| {
 
 struct Args {
     addrs: Vec<SocketAddr>,
-    #[allow(dead_code)] // only used on unix, so dead code on windows
+    #[cfg_attr(target_os = "windows", allow(dead_code))] // only used on unix, so dead code on windows
```
Comment (Johann150) [inline src/main.rs:162]: CI says to chage the formatting here. I would also suggest changing to the [`windows` configuration option](https://doc.rust-lang.org/reference/conditional-compilation.html#unix-and-windows) instead.

```suggestion
    // only used on unix, so dead code on windows
    #[cfg_attr(windows, allow(dead_code))]
```

**Comment by mbrubeck** [comment]
Instead of allowing dead code, should we make this field `#[cfg(unix)]` so it just doesn't exist on other platforms?

**Comment by Johann150** [comment]
That would also be a nice option, if @w1kee wants to do that. It would require adjusting a few more places in the code.

I wasn't sure if `#[cfg(unix)]` can be used on things like fields in constructors but apparently that works too.

**Comment by Johann150** [comment]
Thanks for your contribution!

**Pull request merged**